middleware.go

```diff
@@ -20,6 +20,28 @@ func (mw MiddlewareFunc) Middleware(handler http.Handler) http.Handler {
 	return mw(handler)
 }
 
+// NewMiddleware helper method which allows creating a middleware from a function which receives the response and request.
+// The typical case would be passing a closure
+func NewMiddleware(mw func(http.ResponseWriter, *http.Request)) MiddlewareFunc {
```

Comment:
The purpose of `NewMiddleware` is not clear to me. IIUC, `NewMiddleware` takes an HTTP handler A as input and produces a `MiddlewareFunc` as output. When the `MiddlewareFunc` is invoked to wrap an HTTP handler B, it produces a new HTTP handler C that invokes A and B in sequence.

I personally fail to see how this can be useful. Can you please explain in more detail the problem that you are trying to solve?